src/complex/Common/TypesUtility.hpp

```diff
@@ -57,4 +59,46 @@ constexpr NumericType GetNumericType() noexcept
     static_assert(AlwaysFalse_v<T>, "complex::GetNumericType: Unsupported type");
   }
 }
+
+inline constexpr usize GetNumericTypeSize(const NumericType& numericType)
+{
+  usize typeSize = 0;
+  switch(numericType)
+  {
+  case NumericType::int8:
+    typeSize = sizeof(int8);
+    break;
+  case NumericType::uint8:
+    typeSize = sizeof(uint8);
+    break;
+  case NumericType::int16:
+    typeSize = sizeof(int16);
+    break;
+  case NumericType::uint16:
+    typeSize = sizeof(uint16);
+    break;
+  case NumericType::int32:
+    typeSize = sizeof(int32);
+    break;
+  case NumericType::uint32:
+    typeSize = sizeof(uint32);
+    break;
+  case NumericType::int64:
+    typeSize = sizeof(int64);
+    break;
+  case NumericType::uint64:
+    typeSize = sizeof(uint64);
+    break;
+  case NumericType::float32:
+    typeSize = sizeof(float32);
+    break;
+  case NumericType::float64:
+    typeSize = sizeof(float64);
+    break;
+  default:
+    throw std::runtime_error("complex::GetNumericTypeSize: Unsupported type");
+  }
+
+  return typeSize;
+}
```

Comment:
```suggestion
inline constexpr usize GetNumericTypeSize(NumericType numericType)
{
  switch(numericType)
  {
  case NumericType::int8:
    return sizeof(int8);
  case NumericType::uint8:
    return sizeof(uint8);
  case NumericType::int16:
    return sizeof(int16);
  case NumericType::uint16:
    return sizeof(uint16);
  case NumericType::int32:
    return sizeof(int32);
  case NumericType::uint32:
    return sizeof(uint32);
  case NumericType::int64:
    return sizeof(int64);
  case NumericType::uint64:
    return sizeof(uint64);
  case NumericType::float32:
    return sizeof(float32);
  case NumericType::float64:
    return sizeof(float64);
  default:
    throw std::runtime_error("complex::GetNumericTypeSize: Unsupported type");
  }
}
```

Enums should be passed by value, and can directly return from case.